acados/ocp_nlp/ocp_nlp_cost_ls.c

```diff
@@ -851,27 +863,42 @@ void ocp_nlp_cost_ls_compute_fun(void *config_, void *dims_, void *model_, void
     int ny = dims->ny;
     int ns = dims->ns;
 
-	// TODO should this overwrite memory->{res,grad,fun,...} (as now) or not ????
-
+    // TODO should this overwrite memory->{res,fun,...} (as now) or not ????
     if (nz > 0)
     {
-		// TODO
-		printf("\nerror: ocp_nlp_cost_ls_compute_fun: not implemented yet for nz>0\n");
-		exit(1);
+        // update y_ref: y_ref_tilde = y_ref + Vz*(dzdx*x + dzdu*u - z)
+        blasfeo_dgemv_t(nx + nu, nz, 1.0, memory->dzdux_tran,
+                0, 0, memory->ux, 0, -1.0, memory->z_alg, 0, &work->tmp_nz, 0);
+
+        blasfeo_dgemv_n(ny, nz, +1.0, &model->Vz,
+                0, 0, &work->tmp_nz, 0, 1.0, &work->y_ref_tilde, 0, &work->y_ref_tilde, 0);
+
+        // res = \tilde{V}_x * x + \tilde{V}_u * u - \tilde{y}_ref
+        blasfeo_dgemv_t(nu + nx, ny, 1.0, &work->Cyt_tilde, 0, 0, memory->ux,
+                0, -1.0, &work->y_ref_tilde, 0, &memory->res, 0);
+
+        // tmp_ny = W * res
+        blasfeo_dsymv_l(ny, ny, 1.0, &model->W, 0, 0, &memory->res,
```

Comment:
you can use the (cached) cholesky factor of W and `dtrmv` as in the `else` code, it's a bit faster